assets/js/src/form_editor/store/controls.jsx

```diff
@@ -215,4 +215,12 @@ export default {
   },
 
   CALL_API,
+
+  ENSURE_BROWSER_URL({ formId }) {
+    let url = select('mailpoet-form-editor').getFormEditorUrl();
+    url = `${url}${formId}`;
+    if (window.location !== url) {
+      window.history.replaceState('Form edit', 'Form edit', url);
```

Comment:
I'm not sure about first 2 parameters here. The first parameter should be a stateObject. According to the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState#parameters) we can pass `null` so I guess it would be safer to go with `null` just to be safe. The second parameter should be a title. Based on the documentation all browsers ignore it except for Safari. So I think it would be better to either pass some translated title or an empty string.